stan/math/opencl/kernels/cumulative_sum.hpp

```diff
@@ -0,0 +1,224 @@
+#ifndef STAN_MATH_OPENCL_KERNELS_CUMULATIVE_SUM_HPP
+#define STAN_MATH_OPENCL_KERNELS_CUMULATIVE_SUM_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_cl.hpp>
+#include <stan/math/opencl/buffer_types.hpp>
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <string>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+// \cond
+static const char *cumulative_sum1_kernel_code = STRINGIFY(
```

Comment:
Could we give them descriptive names as opposed to sum1, sum2? 

If not, maybe a comment on the top of the file with a one line description would be helpful I guess.

Reply:
I am not sure what good names would be. If you have a suggestion, go ahead. They already have comments about what do they do in the doxygen. Would you like to see anything more?